<a href="https://colab.research.google.com/github/ulisesjcaroflores/Lab2/blob/main/Lab2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

avocado_clean = pd.read_csv('/content/avocado-updated-2020.csv')
avocado_clean.head(33000)


date  average_price  total_volume       4046        4225  \
0      2015-01-04           1.22      40873.28    2819.50    28287.42   
1      2015-01-04           1.79       1373.95      57.42      153.88   
2      2015-01-04           1.00     435021.49  364302.39    23821.16   
3      2015-01-04           1.76       3846.69    1500.15      938.35   
4      2015-01-04           1.08     788025.06   53987.31   552906.04   
...           ...            ...           ...        ...         ...   
32995  2020-11-29           1.07    5009846.41  431406.97  2524016.28   
32996  2020-11-29           1.62     215435.78   12492.64    22166.72   
32997  2020-11-29           1.08     446403.65   10336.17   307054.15   
32998  2020-11-29           1.55      20377.28      39.56      316.49   
32999  2020-11-29           0.67     543064.98  239653.06    13805.42   

           4770  total_bags  small_bags  large_bags  xlarge_bags  \
0         49.90     9716.46     9186.93      529.53         0.00   
1          0.00     1162.65     1162.65        0.00         0.00   
2         82.15    46815.79    16707.15    30108.64         0.00   
3          0.00     1408.19     1071.35      336.84         0.00   
4      39995.03   141136.68   137146.07     3990.61         0.00   
...         ...         ...         ...         ...          ...   
32995   7793.81  2046629.35  1311967.11   694415.45     40246.79   
32996     28.66   180747.76   162466.45    18251.14        30.17   
32997    100.52   128912.81    69006.93    55889.22      4016.66   
32998      0.00    20021.23    11279.66     8741.57         0.00   
32999     31.35   289572.31   220834.50    66182.26      2555.55   

               type  year             geography  
0      conventional  2015                Albany  
1           organic  2015                Albany  
2      conventional  2015               Atlanta  
3           organic  2015               Atlanta  
4      conventional  2015  Baltimore/Washington  
...             ...   ...                   ...  
32995  conventional  2020             Northeast  
32996       organic  2020             Northeast  
32997  conventional  2020  Northern New England  
32998       organic  2020  Northern New England  
32999  conventional  2020               Orlando  

[33000 rows x 13 columns]

**This dataset contains the sales of avocados across various U.S. markets over a certain time period ranging from 2014 - 2020. Other metrics includes are average price of the avocado, different sizes through a specific code, the number of avocados sold in a certain size of bag, and whether the avocado was organic or conventional. A interesting detail is the geogrpahy as it is a combination of cities, states, regions, and national total.**

**This helps display the unique values of the geography**

In [ ]:
print(avocado_clean["geography"].unique())

['Albany' 'Atlanta' 'Baltimore/Washington' 'Boise' 'Boston'
 'Buffalo/Rochester' 'California' 'Charlotte' 'Chicago'
 'Cincinnati/Dayton' 'Columbus' 'Dallas/Ft. Worth' 'Denver' 'Detroit'
 'Grand Rapids' 'Great Lakes' 'Harrisburg/Scranton' 'Hartford/Springfield'
 'Houston' 'Indianapolis' 'Jacksonville' 'Las Vegas' 'Los Angeles'
 'Louisville' 'Miami/Ft. Lauderdale' 'Midsouth' 'Nashville'
 'New Orleans/Mobile' 'New York' 'Northeast' 'Northern New England'
 'Orlando' 'Philadelphia' 'Phoenix/Tucson' 'Pittsburgh' 'Plains'
 'Portland' 'Raleigh/Greensboro' 'Richmond/Norfolk' 'Roanoke' 'Sacramento'
 'San Diego' 'San Francisco' 'Seattle' 'South Carolina' 'South Central'
 'Southeast' 'Spokane' 'St. Louis' 'Syracuse' 'Tampa' 'Total U.S.' 'West'
 'West Tex/New Mexico']


**Here we are mapping the geogprahy to a specific category: city, state, region, national**

In [ ]:
geo_map = {
    "Total U.S.": "national",
    "California": "state",
    "South Carolina": "state",
    "West": "region",
    "Great Lakes": "region",
    "Midsouth": "region",
    "Northeast": "region",
    "Northern New England": "region",
    "Plains": "region",
    "Southeast": "region",
    "South Central": "region",
    "West Tex/New Mexico": "region"
}

avocado_clean["geo_type"] = avocado_clean["geography"].map(geo_map).fillna("city")
avocado_clean.head()

date  average_price  total_volume       4046       4225      4770  \
0  2015-01-04           1.22      40873.28    2819.50   28287.42     49.90   
1  2015-01-04           1.79       1373.95      57.42     153.88      0.00   
2  2015-01-04           1.00     435021.49  364302.39   23821.16     82.15   
3  2015-01-04           1.76       3846.69    1500.15     938.35      0.00   
4  2015-01-04           1.08     788025.06   53987.31  552906.04  39995.03   

   total_bags  small_bags  large_bags  xlarge_bags          type  year  \
0     9716.46     9186.93      529.53          0.0  conventional  2015   
1     1162.65     1162.65        0.00          0.0       organic  2015   
2    46815.79    16707.15    30108.64          0.0  conventional  2015   
3     1408.19     1071.35      336.84          0.0       organic  2015   
4   141136.68   137146.07     3990.61          0.0  conventional  2015   

              geography geo_type  
0                Albany     city  
1                Albany     city  
2               Atlanta     city  
3               Atlanta     city  
4  Baltimore/Washington     city

**We are renamingthe sizes of avocado based on the PLU codes.**

In [ ]:
avocado_rename = {
    "4046": "small",
    "4225": "large",
    "4770": "extra large"
}

avocado_clean = avocado_clean.rename(columns=avocado_rename)
avocado_clean.head()

date  average_price  total_volume      small      large  extra large  \
0  2015-01-04           1.22      40873.28    2819.50   28287.42        49.90   
1  2015-01-04           1.79       1373.95      57.42     153.88         0.00   
2  2015-01-04           1.00     435021.49  364302.39   23821.16        82.15   
3  2015-01-04           1.76       3846.69    1500.15     938.35         0.00   
4  2015-01-04           1.08     788025.06   53987.31  552906.04     39995.03   

   total_bags  small_bags  large_bags  xlarge_bags          type  year  \
0     9716.46     9186.93      529.53          0.0  conventional  2015   
1     1162.65     1162.65        0.00          0.0       organic  2015   
2    46815.79    16707.15    30108.64          0.0  conventional  2015   
3     1408.19     1071.35      336.84          0.0       organic  2015   
4   141136.68   137146.07     3990.61          0.0  conventional  2015   

              geography geo_type  
0                Albany     city  
1                Albany     city  
2               Atlanta     city  
3               Atlanta     city  
4  Baltimore/Washington     city

**We are looking for the region that has the highest sales of organic small hass avocados in 2017**

In [ ]:
avocado_clean = avocado_clean[(avocado_clean['type'] == 'organic') & (avocado_clean['year'] == 2017)]
max_small_sold = avocado_clean['small'].max()
avocado_clean[avocado_clean['small'] == avocado_clean['small'].max()]

date  average_price  total_volume      small      large  \
12090  2017-02-19           1.31    1634877.11  250591.21  632101.53   

       extra large  total_bags  small_bags  large_bags  xlarge_bags     type  \
12090      3295.09   748889.28   433926.87   314962.41          0.0  organic   

       year   geography  geo_type  
12090  2017  Total U.S.  national

**We found that the Total U.S. had the highest sales of organic small hass avocados in 2017 with a value of 250591.21**